pandas/core/frame.py

```diff
@@ -2717,7 +2717,7 @@ def to_feather(self, path: FilePath | WriteBuffer[bytes], **kwargs) -> None:
     )
     def to_markdown(
         self,
-        buf: IO[str] | str | None = None,
+        buf: FilePath | BaseBuffer | None = None,
```

Comment:
Can you please replace `BaseBuffer` with `WriteBuffer[str]`? `BaseBuffer` does not require `.write`. I think `to_markdown` needs `.write` to accept strings.

Reply:
Thanks for the fast feedback. Good point. I pushed a new version.